### PR TITLE
firefox: add handlers.json configuration

### PR DESCRIPTION
**Pull request by kugland**
### Description

Adds support for configuring Firefox's handlers.json file to manage MIME type and URL scheme handlers declaratively (at `programs.firefox.profiles.<profile>.handlers`). Handlers control how Firefox opens files and protocols (e.g., PDF
viewers, mailto handlers).

The implementation validates handler configurations and generates properly formatted JSON output. Includes comprehensive test coverage for edge cases and assertions.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `nix fmt` or
      `nix-shell -p treefmt nixfmt deadnix keep-sorted nixf-diagnose --run treefmt`.

- [x] Code tested through `nix run .#tests -- test-all` or
      `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

  ```
  {component}: {description}

  {long description}
  ```

  See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/a51598236f23c89e59ee77eb8e0614358b0e896c/modules/programs/lesspipe.nix#L11).
  - [x] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)
  - [x] Basic tests added. See [Tests](https://nix-community.github.io/home-manager/index.xhtml#sec-tests)

- If this PR adds an exciting new feature or contains a breaking change.
  - [ ] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)

### CODE REVIEW

**Comment by kugland** [comment]
Reopened with the same patch as https://github.com/nix-community/home-manager/pull/8325, as asked.

**Comment by kugland** [comment]
@khaneliman @rycee

**Comment by kugland** [comment]
Will close https://github.com/nix-community/home-manager/issues/5617

File: modules/programs/firefox/profiles/handlers.nix [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                      uriTemplate = null;
+                    }
+                  )
+                );
+              message = "'${cfgName}': an empty handler can only be used when there are additional handlers.";
+            }
+          ]
+        ) cfg
+      )
+    );
```
Comment (khaneliman) [inline modules/programs/firefox/profiles/handlers.nix]: Not sure how I feel about having to maintain all these assertions in Home Manager. Where did these rules come from? Would be helpful to see some kind of documentation or even link to the documentation instead for users.

Comment (kugland) [reply modules/programs/firefox/profiles/handlers.nix]: Ok. So I should remove these checks and add some documentation?

Comment (khaneliman) [reply modules/programs/firefox/profiles/handlers.nix]: IMO, yes. I'd just not like our module to be maintaining enforcement of structure of a config like this. We just want to provide the tools necessary to configure and link to upstream documentation about how to configure. We want the onus on maintaining docs to the developers upstream.

Comment (kugland) [reply modules/programs/firefox/profiles/handlers.nix]: Done. Removed all assertions and all tests related to those assertions (only one test remains, that checks if the file is written). Added a link to an example `handlers.json` in Firefox’s source tree.
